sql/sql_connect.cc

```diff
@@ -985,7 +985,7 @@ static int check_connection(THD *thd)
                       /* See RFC 5737, 192.0.2.0/24 is reserved. */
                       const char* fake= "192.0.2.4";
                       inet_pton(AF_INET,fake, ip4);
-                      strcpy(ip, fake);
+                      snprintf(ip, sizeof(ip), "%s", fake);
```

Comment:
is a const char copy, compilers can detect length, esp with `ip` bloated to `NI_MAXHOST` (man 3 getnameinfo). Likewise below.